src/main/java/io/cdap/delta/bigquery/MultiGCSWriter.java

```diff
@@ -76,35 +82,71 @@ public synchronized void write(Sequenced<DMLEvent> sequencedEvent) {
     }
   }
 
-  public synchronized Collection<TableBlob> flush() throws IOException {
-    List<TableBlob> writtenObjects = new ArrayList<>(objects.size());
-    IOException error = null;
+  public synchronized Collection<TableBlob> flush() throws IOException, InterruptedException {
+    List<TableBlob> writtenObjects = Collections.synchronizedList(new ArrayList<>(objects.size()));
```

Comment:
Instead of having a separate list, you can have the `Callable` return the `TableBlob` object, which you can get through the future.

Reply:
Sure. will do.